### ISSUE TITLE
`options.preloadHoveredLinks` incorrectly set up

### ISSUE THREAD

**Issue by jaxtheking** [swup/preload-plugin#113]
**Description of the issue**

It seems the `preloadHoveredLinks` option has not been set up correctly. If it's set to `false`, links are still preloaded on hover.
This is because the mouse/touch/focus event delegates are created regardless of this option.

Additionally, when set to `false`, it seems the option disables the preloading of links with the [data-swup-preload] attribute instead.
I don't believe this was intentional as the docs clearly state the preloading of these links should happen automatically.

Perhaps totally pointless, but this option could be optionally set as an object (similarly to `preloadVisibleLinks`) to enable/disable
preloading on mouse/touch/focus independently:
```
new SwupPreloadPlugin({
  preloadHoveredLinks: {
    /** Enable preloading on mouse enter */
    mouse: true,
    /** Enable preloading on touch start */
    touch: true,
    /** Enable preloading on focus */
    focus: true
  }
})
```

**How to reproduce the issue**

Simply set `preloadHoveredLinks: false`, open the Network tab in developer tools and see links still being preloaded as they're hovered on. Also, links with the [data-swup-preload] attribute will not be preloaded.

**Before creating this issue...**

- [x] Have you provided all helpful information available?
- [x] Have you checked closed issues for similar/related problems?
- [ ] Have you considered creating a demo so we can help you better?

**Comment by hirasso** [comment]
@jaxtheking Thanks for the report! I can reproduce both issues you are describing. We'll first focus on fixing this before thinking about your proposition with the object for `preloadVisibleLinks`.

**Comment by hirasso** [comment]
@jaxtheking could you describe a use case where you think the object for `preloadHoveredLinks` would come in handy?

**Comment by hirasso** [comment]
@jaxtheking We just released version [3.2.7](https://github.com/swup/preload-plugin/releases/tag/3.2.7) with a fix. Could you check if it's working as expected now? Also, thanks again for using the option and reporting the bug!

**Comment by jaxtheking** [comment]
@hirasso Actually I ended up not using the plugin (all I needed was caching the initial page, which I have done manually), but in the future I might as the plugin has a strong use case.
By looking at the new code, it looks like it's going to work as intended now; thanks for the quick response.

TBH I don't have a use case for the object for `preloadHoveredLinks`. I did admit it was perhaps useless but I thought since you had already implemented such mechanism, it'd be good to give people the choice. Thinking about it again, maybe the options in the object for `preloadVisibleLinks` are all it's needed!
